src/NSwag.Commands/Commands/SwaggerGeneration/TypesToJsonSchemaCommand.cs

```diff
@@ -0,0 +1,153 @@
+            "the object (otherwise allOf/oneOf with $ref is used, default: false).")]
+        public bool AllowReferencesWithProperties
+        {
+            get { return Settings.AllowReferencesWithProperties; }
+            set { Settings.AllowReferencesWithProperties = value; }
+        }
+
+        [Argument(Name = "GenerateKnownTypes", IsRequired = false, Description = "Generate schemas for types in KnownTypeAttribute attributes (default: true).")]
+        public bool GenerateKnownTypes
+        {
+            get { return Settings.GenerateKnownTypes; }
+            set { Settings.GenerateKnownTypes = value; }
+        }
+
+        [Argument(Name = "GenerateXmlObjects", IsRequired = false, Description = "Generate xmlObject representation for definitions (default: false).")]
+        public bool GenerateXmlObjects
+        {
+            get { return Settings.GenerateXmlObjects; }
+            set { Settings.GenerateXmlObjects = value; }
+        }
+
+        [Argument(Name = nameof(AllowAdditionalProperties), IsRequired = false, Description = "If `additionalProperties` on all possible definitions and references are allowed (default: false).")]
+        public bool AllowAdditionalProperties { get; set; }
+
+        protected override async Task<string> RunIsolatedAsync(AssemblyLoader.AssemblyLoader assemblyLoader)
+        {
+            var document = new NJsonSchema.JsonSchema4();
+            var generator = new JsonSchemaGenerator(Settings);
+            var schemaResolver = new JsonSchemaResolver(document, Settings);
+
+#if FullNet
+            var assemblies = PathUtilities.ExpandFileWildcards(AssemblyPaths)
+                .Select(path => Assembly.LoadFrom(path)).ToArray();
+#else
+            var currentDirectory = await DynamicApis.DirectoryGetCurrentDirectoryAsync().ConfigureAwait(false);
+            var assemblies = PathUtilities.ExpandFileWildcards(AssemblyPaths)
+                .Select(path => assemblyLoader.Context.LoadFromAssemblyPath(PathUtilities.MakeAbsolutePath(path, currentDirectory))).ToArray();
+#endif
+
+            var allExportedClassNames = assemblies.SelectMany(a => a.ExportedTypes).Select(t => t.FullName).ToList();
+            var matchedClassNames = ClassNames
+                .SelectMany(n => PathUtilities.FindWildcardMatches(n, allExportedClassNames, '.'))
+                .Distinct();
+
+            foreach (var className in matchedClassNames)
+            {
+                var type = assemblies.Select(a => a.GetType(className)).FirstOrDefault(t => t != null);
+                await generator.GenerateAsync(type, schemaResolver).ConfigureAwait(false);
+            }
+
+            document.AllOf.Add(new JsonSchema4
+            {
+                Reference = document.Definitions.First().Value
+            });
+
+            SetAllowAdditionalProperties(document);
+
+            return document.ToJson();
+        }
+
+        private void SetAllowAdditionalProperties(JsonSchema4 document)
```

Comment:
This should be an option in the JSON Schema generator itself => create a PR in NJsonSchema

Reply:
did pr, this pull after that pull